src/fkooman/VootProvider/LdapVootStorage.php

```diff
@@ -167,8 +191,19 @@ public function isMemberOf($resourceOwnerId, $startIndex = null, $count = null)
         $userDn = $this->getUserDn($resourceOwnerId);
 
         $userGroups = array();
+        
+        $groupsProvider = $this->config->s('LdapVootStorage')->l('groupsProvider');
+    
         /* get the groups the user is a member of */
-        $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $userDn . ')';
+        switch ($groupsProvider):
+            case "posixgroup":
+                $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $resourceOwnerId . ')';
+            break;
+            default:
+                $filter = '(' . $this->config->s('LdapVootStorage')->l('memberAttribute') . '=' . $userDn . ')';
+            break;
+        endswitch;
```

Comment:
The idea behind using  switch is that you have the possibility to add other group providers more easily. I think this is also somewhat personal to choose between the two.